iree/tools/translate_main.cc

```diff
@@ -60,13 +60,14 @@ static llvm::cl::opt<bool> splitInputFile(
 int main(int argc, char **argv) {
   llvm::InitLLVM y(argc, argv);
 
-  mlir::registerMlirDialects();
+  mlir::DialectRegistry registry;
+  mlir::registerMlirDialects(registry);
 #ifdef IREE_HAVE_EMITC_DIALECT
-  mlir::registerEmitCDialect();
+  mlir::registerEmitCDialect(registry);
 #endif  // IREE_HAVE_EMITC_DIALECT
-  mlir::registerXLADialects();
-  mlir::iree_compiler::registerIreeDialects();
-  mlir::iree_compiler::registerIreeCompilerModuleDialects();
+  mlir::registerXLADialects(registry);
+  mlir::iree_compiler::registerIreeDialects(registry);
+  mlir::iree_compiler::registerIreeCompilerModuleDialects(registry);
```

Comment:
You shouldn't need any of theses if the translation functions are handling it correctly.

Reply:
Hmmm we seemed to need them before. Maybe something has changed? My understanding from passes was that they were only responsible for declaring the dialects they *output*, but the main binary was still responsible for saying any valid input dialects. Is it different for translations? I just tested it out and was able to find a few places where our translations fail and I can specify a dialect in their registration to fix it.

Since these were already here, I'll fix this in a followup.

Reply:
These were here before because of the global registry and no mechanism on the translation for declaring their dependencies.